### ISSUE TITLE
Add a setting to enable red squiggles for "single file" IntelliSense

### ISSUE THREAD

**Issue by zhouhangzooo** [microsoft/vscode-cpptools#3171]
When writing c++, if it is wrong and there is no red tilde prompt, the red tilde will appear on another computer.

**Comment by bobbrow** [comment]
We turn off the red tildes in a few cases:
1. If you have the following setting: `"C_Cpp.intelliSenseEngine": "Tag Parser"`
2. If the extension is unable to resolve one of your `#include` statements
3. If you opened a C/C++ file without opening a folder.

**Comment by zhouhangzooo** [comment]
> We turn off the red tildes in a few cases:
> 
> If you have the following setting: "C_Cpp.intelliSenseEngine": "Tag Parser"
> If the extension is unable to resolve one of your #include statements
> If you opened a C/C++ file without opening a folder.



> We turn off the red tildes in a few cases:
> 
> If you have the following setting: "C_Cpp.intelliSenseEngine": "Tag Parser"
> If the extension is unable to resolve one of your #include statements
> If you opened a C/C++ file without opening a folder.

Thank you very much. If I want to customize this prompt function, for example, I want to create a new test. CPP file without opening the folder. Please give me some advice.

**Comment by sean-mcmanus** [comment]
I don't think we have expose a way to enable squiggles in the Open File scenario. We would have to add a setting to enable it. I think we disabled it because users complained about squiggles in the open file scenario, since in that scenario there is no c_cpp_properties.json to change includePath, but if users have C_Cpp.default.* settings then maybe we should enable squiggles with Open File? There could be other reasons we disabled it too that I don't remember.

**Comment by zhouhangzooo** [comment]
> I don't think we have expose a way to enable squiggles in the Open File scenario. We would have to add a setting to enable it. I think we disabled it because users complained about squiggles in the open file scenario, since in that scenario there is no c_cpp_properties.json to change includePath, but if users have C_Cpp.default.* settings then maybe we should enable squiggles with Open File? There could be other reasons we disabled it too that I don't remember.

I'm sorry I didn't say it in detail, but if I open a C/ C ++ file and don't open a folder, I want it to show an error (red tilde). Where should I modify it?I really need to use your cpptools plug-in for extension, and I hope you can give me more help and Suggestions.

**Comment by bobbrow** [comment]
Currently, the only workaround is to open a folder.  You can open an empty folder if you don't want to open the folder where the code file exists.  Then you can open any code file you want and you should get the red tildes.

We can use this issue to track adding a setting to turn on the squiggles for files when no folder is open.

**Comment by saeidtafazzol** [comment]
> Currently, the only workaround is to open a folder. You can open an empty folder if you don't want to open the folder where the code file exists. Then you can open any code file you want and you should get the red tildes.
> 
> We can use this issue to track adding a setting to turn on the squiggles for files when no folder is open.

Empty folder worked for me. Thanks

**Comment by bobbrow** [comment]
A while ago we added support for red squiggles in the single file (no workspace folder) case when there are no issues resolving `#include` statements. I don't recall which version exactly, but the feature is available in the most recent version of the extension.
